sw/host/tests/rom/e2e_bootstrap_entry/src/main.rs

```diff
@@ -0,0 +1,161 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, Result};
+use regex::Regex;
+use std::time::Duration;
+use structopt::StructOpt;
+
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::spiflash::{BlockEraseSize, SpiFlash, SupportedAddressModes, WriteGranularity};
+use opentitanlib::test_utils::init::InitializeTest;
+use opentitanlib::uart::console::{ExitStatus, UartConsole};
+use opentitanlib::util::parse_int::ParseInt;
+
+#[derive(Debug, StructOpt)]
+struct Opts {
+    #[structopt(flatten)]
+    init: InitializeTest,
+
+    #[structopt(
+        long, parse(try_from_str=humantime::parse_duration),
+        default_value = "5s",
+        help = "Bootstrap detection timeout",
+    )]
+    timeout: Duration,
+
+    #[structopt(
+        long, parse(try_from_str=usize::from_str),
+        default_value = "12",
+        help = "JEDEC page of manufacturer",
+    )]
+    jedec_page: usize,
+
+    #[structopt(
+        long, parse(try_from_str=u8::from_str),
+        default_value = "0xEF",
+        help = "JEDEC ID of manufacturer",
+    )]
+    jedec_id: u8,
+
+    #[structopt(
+        long, parse(try_from_str=u8::from_str),
+        default_value = "0x08",
+        help = "JEDEC manufacturer product ID",
+    )]
+    jedec_product: u8,
+
+    #[structopt(
+        long, parse(try_from_str=u32::from_str),
+        default_value = "0x100000",
```

Comment:
qq: does parse support separators for readability, i.e. `0x10_0000`?

Reply:
The `ParseInt` trait uses `from_str_radix` to convert the integer.  `from_str_radix` does not appear to support separators:
```
error: Invalid value for '--flash-size <flash-size>': invalid digit found in string
```